sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs

```diff
@@ -585,8 +585,8 @@ public static partial class ComputeExtensions
         public static Azure.ResourceManager.Compute.CapacityReservationGroupResource GetCapacityReservationGroupResource(this Azure.ResourceManager.ArmClient client, Azure.Core.ResourceIdentifier id) { throw null; }
         public static Azure.ResourceManager.Compute.CapacityReservationGroupCollection GetCapacityReservationGroups(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource) { throw null; }
-        public static Azure.Pageable<Azure.ResourceManager.Compute.CapacityReservationGroupResource> GetCapacityReservationGroups(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand? expand = default(Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public static Azure.AsyncPageable<Azure.ResourceManager.Compute.CapacityReservationGroupResource> GetCapacityReservationGroupsAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand? expand = default(Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Pageable<Azure.ResourceManager.Compute.CapacityReservationGroupResource> GetCapacityReservationGroups(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand? expand = default(Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand?), Azure.ResourceManager.Compute.Models.ResourceIdOptionsForGetCapacityReservationGroup? resourceIdsOnly = default(Azure.ResourceManager.Compute.Models.ResourceIdOptionsForGetCapacityReservationGroup?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.AsyncPageable<Azure.ResourceManager.Compute.CapacityReservationGroupResource> GetCapacityReservationGroupsAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand? expand = default(Azure.ResourceManager.Compute.Models.CapacityReservationGroupGetExpand?), Azure.ResourceManager.Compute.Models.ResourceIdOptionsForGetCapacityReservationGroup? resourceIdsOnly = default(Azure.ResourceManager.Compute.Models.ResourceIdOptionsForGetCapacityReservationGroup?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
we are adding a parameter in the middle of the parameter list, which is a breaking change in SDK's API perspective despite adding an optional parameter in swaggers are not.
We have to mitigate this "breaking change" manually. But considering this is an extension method, and we have a pattern in implementing extension methods to enable mocking for them, this mitigation takes several steps (which is complicated because this method is an extension):
### step 1
create a partial class for `MockableComputeSubscriptionResource` with the same namespace as the existing class (`Azure.ResourceManager.Compute.Mocking`) with the following code (FYI I added comments to illustrate the purpose of doing that, we do not have to have the same comment in your code):
```csharp
namespace Azure.ResourceManager.Compute.Mocking
{
	public partial class MockableComputeSubscriptionResource
	{
		// <copy the xml doc here from the original method, otherwise we will have compilation errors>
		public virtual AsyncPageable<CapacityReservationGroupResource> GetCapacityReservationGroupsAsync(CapacityReservationGroupGetExpand? expand, CancellationToken cancellationToken) // this is the same method signature as we have in the latest stable version, but we change every parameter to required to avoid ambiguous call
		{
			return GetCapacityReservationGroupsAsync(expand, null, cancellationToken); // we just redirect the call to its latest version without the extra parameter
		}
	}
}
```
### step 2
create a partial class for `ComputeExtensions` with the same namespace as the existing class (`Azure.ResourceManager.Compute`) with the following code:
```csharp
namespace Azure.ResourceManager.Compute
{
	public static partial class ComputeExtensions
	{
		/// <copy the xml doc here from the original method, otherwise we will have compilation errors>
		public static AsyncPageable<CapacityReservationGroupResource> GetCapacityReservationGroupsAsync(this SubscriptionResource subscriptionResource, CapacityReservationGroupGetExpand? expand, CancellationToken cancellationToken)
	        {
	            Argument.AssertNotNull(subscriptionResource, nameof(subscriptionResource));
	
	            return GetMockableComputeSubscriptionResource(subscriptionResource).GetCapacityReservationGroupsAsync(expand, cancellationToken); // this is important for mocking, we need to call the method with exactly same signature, otherwise we break the existing mocking code.
	        }
	}
}
```

This might look complicated because we have to keep the functionalities of this method unchanged, and in the meantime, we want to keep the existing mocking to work correctly.
Please let me know if we have other issues after you applied these because I am hand-writing these code without an IDE there might be typos or mistakes, etc...

Reply:
I made those changes and that resolved the issue.

Thanks!